tests/test_binexport_accessors.py

```diff
@@ -0,0 +1,300 @@
+                "bytes": m["bytes"].strip(),
+                "mnemonic": m["mnemonic"],
+                "operands": [e for e in [m["operand1"], m["operand2"], m["operand3"]] if e is not None],
+            }
+        else:
+            logger.debug("No match\t%s", line)
+    return dd
+
+
+BE2_EXTRACTOR = fixtures.get_binexport_extractor(
+    CD
+    / "data"
+    / "binexport2"
+    / "c7f38027552a3eca84e2bfc846ac1307fbf98657545426bb93a2d63555cbb486.elf_.ghidra.BinExport"
+)
+PARSED_DISASM = _parse_ghidra_disassembly(GHIDRA_DISASSEMBLY)
+
+
+def test_instruction_bytes():
+    # more a data sanity check here as we don't test our code
+    for addr, de in PARSED_DISASM.items():
+        insn = BE2_EXTRACTOR.idx.get_instruction_by_address(addr)
+        assert insn.raw_bytes == bytes.fromhex(de["bytes"])
+
+
+def test_get_instruction_mnemonic():
+    for addr, de in PARSED_DISASM.items():
+        insn = BE2_EXTRACTOR.idx.get_instruction_by_address(addr)
+        assert get_instruction_mnemonic(BE2_EXTRACTOR.be2, insn) == de["mnemonic"]
+
+
+def test_get_instruction_operands_count():
+    for addr, de in PARSED_DISASM.items():
+        insn = BE2_EXTRACTOR.idx.get_instruction_by_address(addr)
+        ops = get_instruction_operands(BE2_EXTRACTOR.be2, insn)
+        # this line is not properly parsed from the Ghidra disassembly using the current regex
+        # 00210184 02 68 61 38     ldrb       w2,[x0, x1, LSL ]
+        if addr == 0x210184:
+            assert len(ops) == 2
+        else:
+            assert len(ops) == len(de["operands"])
+
+
+@pytest.mark.parametrize(
+    "addr,op_expressions",
+    [
+        # 00210158 20 00 80 d2     mov        x0,#0x1
+        (0x210158, ("x0", "#0x1")),
+        # 0021015c a1 02 00 58     ldr        x1=>helloWorldStr,DAT_002101b0
+        (0x21015C, ("x1", "DAT_002101b0")),
+        # 00210184 02 68 61 38     ldrb       w2,[x0, x1, LSL ]
+        (0x210184, ("w2", "[x0, x1, LSL ]")),
+        # 00210190 21 04 00 91     add        x1,x1,#0x1
+        (0x210190, ("x1", "x1", "#0x1")),
+    ],
+)
+def test_get_operand_expressions(addr, op_expressions):
+    insn = BE2_EXTRACTOR.idx.get_instruction_by_address(addr)
+    ops = get_instruction_operands(BE2_EXTRACTOR.be2, insn)
+    for i, op in enumerate(ops):
+        exps = get_operand_expressions(BE2_EXTRACTOR.be2, op)
+        assert len(exps) == 1
+        assert exps[0].symbol == op_expressions[i]
+
+
+@pytest.mark.parametrize(
+    "addr,reg_expressions",
+    [
+        # 00210158 20 00 80 d2     mov        x0,#0x1
+        (0x210158, ("x0", None)),
+        # 0021015c a1 02 00 58     ldr        x1=>helloWorldStr,DAT_002101b0
+        (0x21015C, ("x1", None)),
+        # 00210184 02 68 61 38     ldrb       w2,[x0, x1, LSL ]
+        (0x210184, ("w2", None)),
+        # 00210190 21 04 00 91     add        x1,x1,#0x1
+        (0x210190, ("x1", "x1", None)),
+    ],
+)
+def _TODO_test_get_operand_register_expression(addr, reg_expressions):
```

Comment:
@mike-hunhoff could you take a look at these to verify I understand correctly how the respective `get_` functions are supposed to work? these `_TODO_` tests here fail currently.

Reply:
`get_operand_register_expression` returns `Optional[BinExport2.Expression]` so you need to compare against `reg_exp.symbol`.

see https://github.com/mandiant/capa/blob/c662176e38920b741dccca50637a664e9c94664d/capa/features/extractors/binexport2/helpers.py#L198-L203

Reply:
I only see None being returned from get_operand_register_expression